eventuate-local-java-cdc-connector-unified/src/main/java/io/eventuate/local/unified/cdc/pipeline/CdcMessageCleanerConfigurator.java

```diff
@@ -136,6 +145,40 @@ private ConnectionInfo createCustomCleanerConnectionInfo(MessageCleanerPropertie
   private EventuateSchema createEventuateSchema(String schema) {
     return new EventuateSchema(schema == null ? EventuateSchema.DEFAULT_SCHEMA : schema);
   }
+
+  //SPRING PARSES INTERVAL_IN_SECONDS as {interval={in={seconds=1}}}
+  private Map<String, Object> reconstructProperties(Map<String, Object> properties) {
+    List<Pair<String, Object>> propertyList = properties
+            .entrySet()
+            .stream()
+            .map(e -> Pair.of(e.getKey(), e.getValue()))
+            .collect(Collectors.toList());
+
+    return reconstructProperties(propertyList)
+            .stream()
+            .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
+  }
+
+  private List<Pair<String, Object>> reconstructProperties(List<Pair<String, Object>> properties) {
+    List<Pair<String, Object>> newProperties = new ArrayList<>();
+
+    for (Pair<String, Object> p : properties) {
+      if (p.getValue() instanceof Map) {
+        Map<String, Object> value = (Map<String, Object>) p.getValue();
+        for (String key : value.keySet()) {
+          String newKey = p.getKey().concat(key);
+          Object newValue = value.get(key);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.